### PR TITLE
Refactor resize_image to support multiple target sizes

### PR DESCRIPTION
**Pull request by ralsina**
This could eventually lead to supporting things like #3338

In the meantime, gallery rendering in my site takes ~~112~~ 260 seconds when it takes 364 using master.

Still there are opportunities for improving performance:

* EXIF cleanup could be done only once per loop
* Probably something else

But this is not so invasive and gives a large perf improvement, so let's merge it if it's ok.

### CODE REVIEW

File: nikola/image_processing.py [hunk 1]
```diff
@@ -81,69 +81,94 @@ def filter_exif(self, exif, whitelist):
...
+        * Either max_size or max_sizes should be set
+        * Either dst or dst_paths should be set
+        """
+
+        if dst_paths is None:
+            dst_paths = [dst]
+        if max_sizes is None:
+            max_sizes = [max_size]
+        if len(max_sizes) != len(dst_paths):
+            raise ValueError(f'resize_image called with incompatible arguments: {dst_paths} / {max_sizes}')
```
Comment (Kwpolska) [inline nikola/image_processing.py:105]: We still support 3.5, which does not have f-strings.

Comment (ralsina) [reply nikola/image_processing.py:105]: Should we? Changing it anyway :-)

Comment (Kwpolska) [reply nikola/image_processing.py]: 3.5 is generally usable, and Ubuntu 16.04 is still supported.

Comment (ralsina) [reply nikola/image_processing.py]: Oh, well. Changed.

**Comment by ralsina** [comment]
Marking again as WIP because the delta on the demo site looks suspicious.

**Comment by ralsina** [comment]
Locally there is no difference in the output between this and master :shrug:

**Comment by ralsina** [comment]
Ok, so this PR is making all images small for some reason. Fixing.

**Comment by ralsina** [comment]
This PR is done, review much appreciated @Kwpolska ... in the end the performance improvement is a modest 25% in my site, but it's something :-)

File: nikola/image_processing.py [hunk 2]
```diff
@@ -81,111 +81,132 @@ def filter_exif(self, exif, whitelist):
...
+                _im = _im.transpose(Image.ROTATE_270)
             elif value in (7, 8):
-                im = im.transpose(Image.ROTATE_90)
+                _im = _im.transpose(Image.ROTATE_90)
             if value in (2, 4, 5, 7):
-                im = im.transpose(Image.FLIP_LEFT_RIGHT)
+                _im = _im.transpose(Image.FLIP_LEFT_RIGHT)
             exif['0th'][piexif.ImageIFD.Orientation] = 1
+            exif = self.filter_exif(exif, exif_whitelist)
+        except KeyError:
```
Comment (Kwpolska) [inline nikola/image_processing.py:129]: Is that KeyError meant to be coming from `_im.info["exif"]`? We could replace it with an `'exif' in _im.info` or `_im.info.get('exif')`. Same applies to `exif['0th']`.

Comment (ralsina) [reply nikola/image_processing.py:129]: Probably (who can remember? not me :-D) will change.

Comment (ralsina) [reply nikola/image_processing.py]: Done

**Pull request merged**